### PR TITLE
downloading bundles directly into GridFS using streams;

### PR DESCRIPTION
**Pull request by spherefoundry**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] Automatic tests added/update
- [x] Automatic tests pass
- [x] Software runs (locally or hosted)
- [x] Sanity check passed
- [x] Self-review passed

### CODE REVIEW

File: src/services/data_model_engine.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {NotFoundError, PermissionError, ValidationError} from '../errors/errors'
 import {getTimestamp} from '../utils/time_utils';
 import {pick, put} from '../utils/dict_utils';
 import allPermissions from '../utils/all_permissions';
+const pipeline = require('util').promisify(require('stream').pipeline);
```
Comment (vanruch) [inline src/services/data_model_engine.js]: I don't have anything against it, but if we have imports everywhere, is it fine to use require in that one place?

File: src/services/data_model_engine.js [hunk 2]
```diff
@@ -236,23 +237,29 @@ export default class DataModelEngine {
...
-    if (!bundle) {
+    try {
+      const downloadStream = await this.bundleDownloader.openBundleDownloadStream(nodeUrl, bundleId);
+      const writeStream = await this.bundleRepository.openBundleWriteStream(bundleId, bundleMetadata.storagePeriods);
+      downloadStream.on('error', (err) => {
+        writeStream.abort(err);
+      });
+      await pipeline(downloadStream, writeStream);
+    } catch {
       throw new Error('Could not fetch the bundle from the shelterer');
```
Comment (vanruch) [inline src/services/data_model_engine.js]: This error can be misleading in case exception was thrown anywhere but line 240

File: src/services/data_model_engine.js (hunk 1 above)
Comment (spherefoundry) [reply src/services/data_model_engine.js]: Sadly this a limitation of import. To use imports and achieve the same thing I would have to split it into three lines:
1) import utli <- would also pollute the context with something I don't need otherwise
2) import the unpromisified version of pipeline <- would also pollute context
3) the the promisification and assign it into a const <- the name of the const can conflict with the import in point 2), so one would have to be reassigned.

IMO: using `require` is just simpler and cleaner in this case
